### PR TITLE
[mqtt] issue 5027: the smaller version

### PR DESCRIPTION
**Pull request by jochen314**
Issue #5027 

This version is only extending OnOffValue.

It is easier.
I just not sure, if we need to change others (Color, OpenClose, Percentage, RollerShutter) as well

### CODE REVIEW

**Comment by jochen314** [comment]
The failing build is not caused by this PR, or?
I would not see how.

**Comment by davidgraeff** [comment]
I don't think so. Others: This is ready for a second review.

**Comment by davidgraeff** [comment]
@openhab/2-x-add-ons-maintainers Could someone have a second look, please? :)

File: addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/values/OnOffValue.java [hunk 1]
```diff
@@ -29,28 +29,46 @@
...
+
+    /**
+     * Creates a new SWITCH On/Off value.
+     *
+     * @param onState The ON value string. This will be compared to MQTT messages.
+     * @param offState The OFF value string. This will be compared to MQTT messages.
+     * @param onCommand The ON value string. This will be send in MQTT messages.
+     * @param offCommand The OFF value string. This will be send in MQTT messages.
+     */
+
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/values/OnOffValue.java]: Can you please remove this empty line.

**Comment by cweitkamp** [comment]
@jochen314 Can you fix the conflict?

**Comment by davidgraeff** [comment]
For merge conflicts I recommend the rebase method (rewriting the history + force push). Some people out there say that a base merge is the only right way to go, but personally I do not like the interrupted history line, when viewing with gitk or any other tool.

**Comment by martinvw** [comment]
@wborn I see a travis failure which reminds of recent core changes, are they related?

```

[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.openhab.binding.plclogo 2.5.0.qualifier
[ERROR]   Missing requirement: org.eclipse.xtext.util 2.17.0.v20190304-0545 requires 'osgi.bundle; com.google.guava [21.0.0,22.0.0)' but it could not be found
[ERROR]   Cannot satisfy dependency: org.eclipse.xtext 2.17.0.v20190304-0545 depends on: osgi.bundle; org.eclipse.xtext.util 0.0.0
[ERROR]   Cannot satisfy dependency: org.openhab.binding.plclogo 2.5.0.qualifier depends on: java.package; org.eclipse.smarthome.model.script.actions 0.0.0
[ERROR]   Cannot satisfy dependency: org.openhab.core.model.item 2.5.0.201903150831 depends on: osgi.bundle; org.eclipse.xtext 0.0.0
[ERROR]   Cannot satisfy dependency: org.openhab.core.model.script 2.5.0.201903150823 depends on: osgi.bundle; org.openhab.core.model.item 0.0.0

**Comment by jochen314** [comment]
@davidgraeff You just had me convinced, that I would never force push again ;-)

**Pull request merged**